app/components/MHForm.jsx

```diff
@@ -107,7 +107,9 @@ class MHForm extends React.Component {
             case this.FieldTypes.NUMBER:
             case this.FieldTypes.BOOLEAN:
             case this.FieldTypes.ARRAY:
-                return field.default || defaultValue;
+                return field.default !== undefined
```

Comment:
Shouldn't undefined be false?

Reply:
sometimes we want to return false. a false field.default will never return using the || operator

Reply:
I can dig it